### DISCUSSION TITLE
Configuration file: any way to get a boolean value?

### DISCUSSION THREAD

**Discussion by paulross80** [chalet-org/chalet#252] [Q&A] [answered]
I have a configuration file for my app

```
#ifndef APP1_HEADER_HPP
#define APP1_HEADER_HPP

#include <string_view>

namespace app1
{

static const int v_major = ${meta:versionMajor};
static const int v_minor = ${meta:versionMinor};
static const int v_patch = ${meta:versionPatch};
static const int v_tweak = ${meta:versionTweak};

static const std::string_view configuration = "${configuration}";

} // app1

#endif
```

And it works as expected:

```
#ifndef APP1_HEADER_HPP
#define APP1_HEADER_HPP

#include <string_view>

namespace app1
{

static const int v_major = 1;
static const int v_minor = 0;
static const int v_patch = 0;
static const int v_tweak = 2023;

static const std::string_view configuration = "Debug";

} // app1

#endif
```

Can I add a boolean variable that is true for "Debug" builds only?
I know CMake has `#cmakedefine01`. Not ideal, but close
The idea is to get something like this:

`static const bool debug_mode = true;`

Can I do this in chalet?

**Comment by rewrking** [comment, marked as answer]
Hmmm... You can obviously do this any number of ways, but I'm used to using a preprocessor definition at build time:

```json
"settings:Cxx": {
   "defines[:debug]": [
      "_DEBUG"
   ]
}
```

```cpp
namespace app1::constants
{
#if defined(_DEBUG)
constexpr bool debug = true;
#else
constexpr bool debug = false;
#endif
}
```

So you can do that in the meantime, but yes -  I think there should be some mechanism for boolean values like you describe. I'll have to revisit config files because right now the same one is shared across build configurations, so it's a bug to put the `${configuration}` string into it at the moment. It was mainly designed with versioning in mind.

**Reply by paulross80** [reply]
Well...

```
namespace app1::compile_time
{

static constexpr int v_major = 1;
static constexpr int v_minor = 0;
static constexpr int v_patch = 0;
static constexpr int v_tweak = 2023;

//static const std::string_view configuration = "Debug";

#if defined(_DEBUG)
    static constexpr bool debug = true;
#else
    static constexpr bool debug = false;
#endif

} // app1::compile_time
```

It works for now. Thank you, Andrew!

**Reply by rewrking** [reply]
I added this as a feature for the next release (see #253). You'll be able to write:

```cpp
namespace test
{
constexpr bool isDebug = ${defined:_DEBUG};
}
```

`${configuration}` and other build-specific variables will be valid as well.

**Reply by paulross80** [reply]
great!
